src/pcp/types.rs

```diff
@@ -217,6 +217,133 @@ impl<F: FieldElement> Value for Sum<F> {
     }
 }
 
+/// The histogram type. Each measurement is a non-negative integer and the aggregate is a histogram
+/// approximating the distribution of the measurements.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct Histogram<F> {
+    data: Vec<F>,
+    range_checker: Vec<F>,
+    sum_check_share: F,
+}
+
+impl<F: FieldElement> Histogram<F> {
+    /// Constructs a new histogram input. The values of `buckets` must be strictly increasing.
+    pub fn new(measurement: u64, buckets: &[u64]) -> Result<Self, PcpError> {
+        let mut data = vec![F::zero(); buckets.len() + 1];
+
+        if buckets.len() == 0 {
```

Comment:
nit: elsewhere, `num_buckets` is `u32`, so perhaps you should check that `buckets.len() <= u32.MAX`?

Reply:
Done.